lib/reek/configuration/app_configuration.rb

```diff
@@ -5,89 +5,108 @@ module Reek
   # @api private
   module Configuration
     # @api private
-    class ConfigFileException < StandardError; end
     #
     # Reek's singleton configuration instance.
     #
     # @api private
-    module AppConfiguration
-      NON_SMELL_TYPE_KEYS   = %w(exclude_paths)
-      EXCLUDE_PATHS_KEY     = 'exclude_paths'
-      @configuration        = {}
-      @has_been_initialized = false
+    class AppConfiguration
+      EXCLUDE_PATHS_KEY = 'exclude_paths'
+      attr_reader :exclude_paths, :default_directive, :directory_directives
 
-      class << self
-        attr_reader :configuration
+      # Given this configuration file:
+      #
+      # ---
+      # IrresponsibleModule:
+      #   enabled: false
+      # "app/helpers":
+      #   UtilityFunction:
+      #     enabled: false
+      # exclude_paths:
+      #   - "app/controllers"
+      #
+      # this would result in the following configuration:
+      #
+      # exclude_paths = [Pathname('spec/samples/two_smelly_files')]
+      # default_directive = { Reek::Smells::IrresponsibleModule=>{"enabled"=>false} }
+      # directory_directives = { Pathname("spec/samples/three_clean_files/") =>
+      #                          {Reek::Smells::UtilityFunction=>{"enabled"=>false}}}
+      #
+      #
+      # @param options [OpenStruct]
+      #   e.g.: #<OpenStruct report_format=:text,
+      #                      location_format=:numbers,
+      #                      colored=true,
+      #                      smells_to_detect=[],
+      #                      config_file=#<Pathname:config/defaults.reek>,
+      #                      argv=["lib/reek/spec"]>
+      def initialize(options = nil)
+        self.directory_directives = {}
```

Comment:
>> We also have
@configuration = Configuration::AppConfiguration.new @options
in this PR. :smile:

Yes, I only changed it where it was actually possible without being inconsistent within one file, at least i tried :smile: 
We should agree on one style and then change that in lib/ in one big pull request.